File: AuthenticationServices/src/main/java/com/artauction/authenticationservices/Repositories/Interfaces/AdminRepo.java
package com.artauction.authenticationservices.Repositories.Interfaces;

import com.artauction.authenticationservices.Entities.Admin;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface AdminRepo extends JpaRepository<Admin, Integer> {
}

File: Project/Backend/src/ArtAuction.Infrastructure/Repositories/AdminRepo.cs
using ArtAuction.Application.Entities;
using ArtAuction.Application.Interfaces.Repositories;
using ArtAuction.Infrastructure.Persistence;
using Microsoft.EntityFrameworkCore;

namespace ArtAuction.Infrastructure.Repositories;

public class AdminRepo : IAdminRepo
{
    // Attributes
    private readonly AppDbContext _dbContext;
    
    // Constructor
    public AdminRepo(AppDbContext dbContext, ISaveChanges saveChanges)
    {
        this._dbContext = dbContext;
    }
    
    // Methods
    public async Task<Admin?> GetAdminById(int adminId)
    {
        return await _dbContext.Admins.AsNoTracking().FirstOrDefaultAsync(admin => admin.Id == adminId);
    }
}